core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3289,6 +3286,11 @@ private FunctionMetadata getFunctionMetadata(FunctionCall functionCall)
             return metadata.getFunctionMetadata(resolvedFunction);
         }
 
+        private List<ColumnMetadata> getColumnMetadatas(QualifiedObjectName tableName)
+        {
+            return metadata.listTableColumns(session, tableName.asQualifiedTablePrefix()).get(tableName);
```

Comment:
Oh man...

Getting column metadata is very important thing in semantic analyzer. Dedicated SPI method seems like a way to go. This method `listTableColumns` has a different purpose. Consider https://github.com/trinodb/trino/issues/6551.

Reply:
Thank you for the link. 
I just try to reuse existing api. And it is quite surprising that it behaves differently than it would be expected.

Reply:
> And it is quite surprising that it behaves differently than it would be expected.

I guess that depends. Once you get more familiar with ideas behind it you will get used to it.